lib/storage/src/content_manager/toc/collection_meta_ops.rs

```diff
@@ -339,6 +341,32 @@ impl TableOfContent {
                     )
                     .await?;
             }
+            ShardTransferOperations::Restart(transfer) => {
+                let transfers = collection.state().await.transfers;
+
+                // Transfer must exist
+                transfer::helpers::validate_transfer_exists(&transfer.key(), &transfers)?;
+
+                // Transfer must have changed configuration
+                if transfers.contains(&transfer) {
+                    return Err(StorageError::bad_request(format!(
+                        "Cannot restart transfer for shard {} from {} to {}, its configuration did not change",
+                        transfer.shard_id, transfer.from, transfer.to,
+                    )));
+                }
+
+                // Abort and start transfer
+                self.handle_transfer(
+                    collection_id.clone(),
+                    ShardTransferOperations::Abort {
+                        transfer: transfer.key(),
+                        reason: "restart transfer".into(),
+                    },
+                )
+                .await?;
+                self.handle_transfer(collection_id, ShardTransferOperations::Start(transfer))
+                    .await?;
```

Comment:
Here I just call abort and start right after each other, as part of a single _operation_. It keeps the implementation very simple.

Maybe there's more to it though.